### ISSUE TITLE
Protocol methods more than 50 lines below the class statement are flagged as placeholders

### ISSUE THREAD

**Issue by CognitiveDecisions** [rsionnach/sloppylint#22]
Follow-up to #6, which fixed the `@abstractmethod` case. The Protocol/ABC case is still open because `_is_likely_protocol_method` scans backwards through source **lines** with a 50-line window:

https://github.com/rsionnach/sloppylint/blob/e0e3ced/src/sloppy/patterns/hallucinations.py#L118

```python
func_line = node.lineno - 1
for i in range(func_line - 1, max(func_line - 50, -1), -1):
```

Any Protocol long enough that a method sits more than 50 lines below `class Foo(Protocol):` loses the guard. Real protocols hit this constantly.

### Repro

```python
from typing import Protocol


class Store(Protocol):
    def near(self) -> str:
        ...

    # ... 60 lines of other methods or comments ...

    def far(self) -> str:
        ...
```

```
$ sloppylint proto.py --format compact
  proto.py:69  ellipsis_placeholder: Placeholder function with ... - implementation needed
```

`near()` at line 5 is correctly skipped. `far()` at line 69 is flagged. Same class, same Protocol, same body.

### Scale

On one real file (a storage Protocol with 49 methods), every method is reported as a placeholder.

### Suggested fix

The AST already knows the enclosing class, so the line scan is not needed. Walking with the parent in hand removes the window entirely:

```python
def _functions(node, prefix, cls):
    for child in ast.iter_child_nodes(node):
        if isinstance(child, ast.ClassDef):
            yield from _functions(child, f"{prefix}{child.name}.", child)
        elif isinstance(child, (ast.FunctionDef, ast.AsyncFunctionDef)):
            yield child, f"{prefix}{child.name}", cls
            yield from _functions(child, f"{prefix}{child.name}.", None)
        else:
            yield from _functions(child, prefix, cls)
```

Then `cls.bases` is checked directly. This also fixes dotted forms (`abc.ABC`) and subscripted ones (`Protocol[T]`) for free, and stops a nested function inside a Protocol method inheriting the guard.

Worth noting the same helper is duplicated character-for-character in three classes (`PassPlaceholder` L89, `EllipsisPlaceholder` L192, `NotImplementedPlaceholder` L302), so the fix currently has to be applied three times.

Found on 0.5.1 (`e0e3ced`).
